data_structures/src/superblock.rs

```diff
@@ -0,0 +1,717 @@
+use crate::chain::{
+    BlockHeader, Bn256PublicKey, Hash, Hashable, PublicKeyHash, SuperBlock, SuperBlockVote,
+};
+use std::collections::HashSet;
+
+use serde::{Deserialize, Serialize};
+
+use witnet_crypto::{hash::Sha256, merkle::merkle_tree_root as crypto_merkle_tree_root};
+
+use failure::_core::iter::Iterator;
+
+/// Possible result of SuperBlockState::add_vote
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum AddSuperBlockVote {
+    /// vote already counted
+    AlreadySeen,
+    /// invalid superblock index
+    InvalidIndex,
+    /// vote corresponding to an ARS member
```

Comment:
`MaybeValid` means that we cannot check the validity of the superblock because we do not have the required ARS state.